src/sources/util/http/prelude.rs

```diff
@@ -119,15 +129,28 @@ pub trait HttpSource: Clone + Send + Sync + 'static {
                             .is_valid(&auth_header)
                             .and_then(|()| decode(&encoding_header, body))
                             .and_then(|body| {
-                                self.build_events(body, headers, query_parameters, path.as_str())
+                                self.build_events(
+                                    body,
+                                    headers.clone(),
+                                    query_parameters.clone(),
```

Comment:
I'd think we could avoid cloning here by the `build_events` not taking ownership of these.

Reply:
That makes this a bit of a refactor--there are two other sources that implement this trait, and they don't have the split off enrichment logic. How about we wait until we've implemented the component spec metrics for them? And that's assuming that this doesn't introduce a performance regression 😄

Reply:
Since this change is introducing a clone for all of the sources that implement this trait (which will decrease performance), I would argue that is a reasonable and correct time to make this change.
The `build_events()` function was previously assuming `build_events()` is the last place these two variables would be used and thus the move was not an issue, but this change alters that.
The changes to make this should be minimal, just changing the function signatures to take a reference and updating the call sites.
You could also make the new `enrich_events()` function take them as references, which would future proof that a bit.

Reply:
Shouldn't/wouldn't we see that performance regression in this PR though, it doesn't appear to be triggering the detector - so I'm not sure it's a hard requirement to include that code change here.

I definitely appreciate the desire to keep things running quickly, so if the change is not very invasive - doing it now would be nice. I don't know if we'd want to block this large-ish PR to do so though.

Reply:
separately- could this PR have been broken up with the testing changes separate to the application of the test?

Reply:
I don't see anything noteworthy in the expanded table - but yeah under five minutes seems reasonable to do here.

Reply:
I am also ofc happy to help, lmk.

Reply:
> separately- could this PR have been broken up with the testing changes separate to the application of the test?

Unfortunately, they were pretty coupled. I had to write the tests to discover that additional test harness changes that were necessary.

Reply:
@neuronull this update should be in the latest changeset.

Reply:
👍  I see it there. thanks for adding that.